### PR TITLE
Fix call to create_publisher after API changed

### PR DESCRIPTION
**Pull request by emersonknapp**
Connects to ros2/rmw#171

Signed-off-by: Emerson Knapp <eknapp@amazon.com>

### CODE REVIEW

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h [hunk 1]
```diff
@@ -1,10 +1,10 @@
 /*
  * Copyright (c) 2008, Willow Garage, Inc.
  * All rights reserved.
- * 
+ *
```
Comment (sloretz) [inline tf2_ros/include/tf2_ros/static_transform_broadcaster.h]: @tfoote this commit has some unrelated whitespace removal. Ok to merge with it?

**Review by sloretz — COMMENTED** [review]
LGTM to be merged after ros2/rclcpp#695 pending question on whitespace

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h (hunk 1 above)
Comment (tfoote) [reply tf2_ros/include/tf2_ros/static_transform_broadcaster.h]: I'd rather not strip the whitespace so we can stay as compatible as possible with the ROS1 repos for merging features across.

**Review by tfoote — APPROVED** [review]
This looks good to me for a simultaneous merge with the rclcpp change. I'm assuming that there will be an integrated CI run for that merge elsewhere.

**Comment by mm318** [comment]
@tfoote I am currently not getting a clean run using the current ros.repos of `ros2/master`. Seeing failures in `demo_nodes_cpp`, `launch`, `launch_ros`, `launch_testing_ros`, `rcl`, `rcl_action`, `rclcpp`, `rclpy`, etc.

**Comment by dirk-thomas** [comment]
> Seeing failures in ...

If you could link to the actual failures we can probably tell if they are known and/or have already been addressed.

**Comment by mm318** [comment]
Sorry, false alarm!

Looks like it was mostly a networking issue. Docker and VPN don't play nice.

I'm still getting quite a few `AttributeError: 'NoneType' object has no attribute 'notify'` and `UnicodeDecodeError: 'ascii' codec can't decode byte 0xXX in position XXXX: ordinal not in range(128)`, but I suspect that is an environment setup issue as well (although my environment has worked before).

**Comment by dirk-thomas** [comment]
> I'm still getting quite a few ...

Again, if you could point to the actual errors, where they happen etc. we might be able to provide more feedback.

**Comment by mm318** [comment]
I am seeing the following stack traces at a couple of places in `rclpy`, `launch`, and `launch_ros`:
```
Traceback (most recent call last):
  File "/opt/ros/crystal/ament_copyright/bin/ament_copyright", line 11, in <module>
    load_entry_point('ament-copyright==0.7.0', 'console_scripts', 'ament_copyright')()
  File "/opt/ros/crystal/ament_copyright/lib/python3.6/site-packages/ament_copyright/main.py", line 119, in main
    file_descriptors[filename] = parse_file(filename)
  File "/opt/ros/crystal/ament_copyright/lib/python3.6/site-packages/ament_copyright/parser.py", line 161, in parse_file
    d.parse()
  File "/opt/ros/crystal/ament_copyright/lib/python3.6/site-packages/ament_copyright/parser.py", line 101, in parse
    self.read()
  File "/opt/ros/crystal/ament_copyright/lib/python3.6/site-packages/ament_copyright/parser.py", line 53, in read
    self.content = h.read()
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1289: ordinal not in range(128)
```
```
Traceback (most recent call last):
  File "/opt/ros/crystal/ament_flake8/bin/ament_flake8", line 11, in <module>
    load_entry_point('ament-flake8==0.7.0', 'console_scripts', 'ament_flake8')()
  File "/opt/ros/crystal/ament_flake8/lib/python3.6/site-packages/ament_flake8/main.py", line 75, in main
    max_line_length=args.linelength)
  File "/opt/ros/crystal/ament_flake8/lib/python3.6/site-packages/ament_flake8/main.py", line 175, in generate_flake8_report
    report.report = style.check_files(paths)
  File "/usr/lib/python3/dist-packages/flake8/api/legacy.py", line 101, in check_files
    self._application.report_errors()
  File "/usr/lib/python3/dist-packages/flake8/main/application.py", line 340, in report_errors
    results = self.file_checker_manager.report()
  File "/usr/lib/python3/dist-packages/flake8/checker.py", line 267, in report
    results_reported += self._handle_results(filename, results)
  File "/usr/lib/python3/dist-packages/flake8/checker.py", line 166, in _handle_results
    physical_line=physical_line,
  File "/usr/lib/python3/dist-packages/flake8/style_guide.py", line 394, in handle_error
    self.listener.notify(error.code, error)
AttributeError: 'NoneType' object has no attribute 'notify'
```

**Comment by dirk-thomas** [comment]
For the first one (`UnicodeDecodeError`) I would speculate that you might not have set the locale properly?

**Comment by clalancette** [comment]
And for the second one, I've seen that when I used the system version of flake8 (not the one from pip).  It's probably a good idea to recheck your installation and environment against https://index.ros.org/doc/ros2/Installation/Linux-Development-Setup/

**Comment by mm318** [comment]
I redid the set locale and `pip install -U flake8 etc.` steps found on the https://index.ros.org/doc/ros2/Installation/Linux-Development-Setup/ page and those issues went away. Thank you, @dirk-thomas and @clalancette!

**Review by wjwwood — APPROVED** [review]

**Comment by dirk-thomas** [comment]
> And for the second one, I've seen that when I used the system version of flake8 (not the one from pip).

This has also been addressed by ament/ament_lint#141 to work well with both versions.
